x/liquidstaking/keeper/tally.go

```diff
@@ -35,6 +35,19 @@ func (k Keeper) GetVoterBalanceByDenom(ctx sdk.Context, votes govtypes.Votes) ma
 	return denomAddrBalanceMap
 }
 
+// GetBTokenSharePerPoolCoinMap creates bTokenSharePerPoolCoinMap of pool coins which is containing target denom to calculate target denom value of farming positions
+func (k Keeper) GetBTokenSharePerPoolCoinMap(ctx sdk.Context, targetDenom string) map[string]sdk.Dec {
+	bTokenSharePerPoolCoinMap := map[string]sdk.Dec{}
+	_ = k.liquidityKeeper.IterateAllPools(ctx, func(pool liquiditytypes.Pool) (stop bool, err error) {
+		bTokenSharePerPoolCoin := k.TokenSharePerPoolCoin(ctx, targetDenom, pool.PoolCoinDenom)
+		if bTokenSharePerPoolCoin.IsPositive() {
+			bTokenSharePerPoolCoinMap[pool.PoolCoinDenom] = bTokenSharePerPoolCoin
+		}
+		return false, nil
+	})
+	return bTokenSharePerPoolCoinMap
+}
+
```

Comment:
Instead of iterating through all pools, how about excluding pools when `targetDenom` is not equal to  `pool.PoolCoinDenom`? And, is it correct that `targetDenom` is always `liquidBondDenom` as the name of the function represents?

Reply:
As @dongsam explained this to me offline, I understand why iterating all pools is necessary. Because it needs to look up pair to know if the `targetDenom` is `QuoteCoinDenom` or `BaseCoinDenom`.